waiter/src/waiter/core.clj

```diff
@@ -187,6 +187,15 @@
                                 (rr/header "server" server-name)))]
       (ru/update-response response add-server-header))))
 
+(defn attach-waiter-api-middleware
+  "Attaches a boolean value for :waiter-api-call? to the response."
+  [handler waiter-request?-fn]
+  (fn attach-waiter-api-middleware-fn [request]
+    (ru/update-response
+      (handler request)
```

Comment:
Can we add it the request map and use that instead of calling `waiter-request?-fn` again downstream?

Reply:
Done. Preserved the `waiter-request?-fn` argument in `ring-handler-factory` to avoid wholesale changes in the unit tests.